In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
data=pd.read_excel('C:/Users/Vishal/Desktop/LU aiml/Data/Linear Regression.xlsx')
data.head()

price  sqft_living  bedrooms  bathrooms  floors
0  221900         1180         3       1.00     1.0
1  538000         2570         3       2.25     2.0
2  180000          770         2       1.00     1.0
3  604000         1960         4       3.00     1.0
4  510000         1680         3       2.00     1.0

# Linear regression model of price vs Floors

In [3]:
floor=data.drop(['sqft_living','bedrooms','bathrooms'],axis=1)
floor.head()

price  floors
0  221900     1.0
1  538000     2.0
2  180000     1.0
3  604000     1.0
4  510000     1.0

In [4]:
x=floor[['floors']]
y=floor[['price']]

In [5]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.25,random_state=2)

In [6]:
from sklearn.linear_model import LinearRegression
linreg=LinearRegression()

In [7]:
linreg.fit(x_train,y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

In [10]:
linreg.coef_

array([[170674.2654533]])

In [11]:
linreg.intercept_

array([283993.57314855])

__Equation is:__

__price = 283993.57314855 + 170674.2654533 (floors)__

In [12]:
ypred=linreg.predict(x_test)

In [13]:
from sklearn.metrics import r2_score, mean_squared_error
r2_score(y_test,ypred)

0.06855333989826096

In [14]:
mean_squared_error(y_test,ypred)

137614630972.6618

# INFERENCE:
R-square value is 0.06855333989826096 means there is 6.8% variance in the target variable.

Totat Error in the model is 137614630972.6618